synthtool/gcp/templates/python_notebooks_testing_pipeline/.cloud-build/execute_notebook_helper.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Methods to run a notebook locally"""
+
+import sys
+import os
+import errno
+import papermill as pm
+import shutil
+
+from utils import util
+from google.cloud.aiplatform import utils
+
+# This script is used to execute a notebook and write out the output notebook.
+
+
+def execute_notebook(
+    notebook_source: str, output_file_or_uri: str, should_log_output: bool,
+):
+    """Execute a single notebook using Papermill"""
+    file_name = os.path.basename(os.path.normpath(notebook_source))
+
+    # Download notebook if it's a GCS URI
+    if notebook_source.startswith("gs://"):
+        # Extract uri components
+        bucket_name, prefix = utils.extract_bucket_and_prefix_from_gcs_path(
+            notebook_source
+        )
+
+        # Download remote notebook to local file system
+        notebook_source = file_name
+        util.download_file(
+            bucket_name=bucket_name, blob_name=prefix, destination_file=notebook_source
+        )
+
+    execution_exception = None
+
+    # Execute notebook
+    try:
+        # Execute notebook
+        pm.execute_notebook(
+            input_path=notebook_source,
+            output_path=notebook_source,
+            progress_bar=should_log_output,
+            request_save_on_cell_execute=should_log_output,
+            log_output=should_log_output,
+            stdout_file=sys.stdout if should_log_output else None,
+            stderr_file=sys.stderr if should_log_output else None,
+        )
+    except Exception as exception:
+        execution_exception = exception
```

Comment:
Please remove the above 2 lines or add a comment to clarify why it is needed along with `try/finally`

https://docs.python.org/3/reference/compound_stmts.html#finally 
> If [finally](https://docs.python.org/3/reference/compound_stmts.html#finally) is present, it specifies a ‘cleanup’ handler. The [try](https://docs.python.org/3/reference/compound_stmts.html#try) clause is executed, including any [except](https://docs.python.org/3/reference/compound_stmts.html#except) and else clauses. If an exception occurs in any of the clauses and is not handled, the exception is temporarily saved. The finally clause is executed. If there is a saved exception it is re-raised at the end of the finally clause.